### DISCUSSION TITLE
Printing Labels

### DISCUSSION THREAD

**Discussion by yodaphone** [inventree/InvenTree#7319] [General]
Hi

I am new to this & have just installed it using docker. 

I imported the demo dataset.

I picked a part and using my camera was able to scan a barcode to that.

Now when i go to print label, i dont see any thing in the "Select Label template" drop down, but i see it in the demo site

![image](https://github.com/inventree/InvenTree/assets/26359077/41197fb8-85cc-4cb5-92f5-64adbfa714cc)

this is from the demo site
![image](https://github.com/inventree/InvenTree/assets/26359077/0fb69ddc-53dc-4fcc-a692-296f1a21756d)

for e.g. i was able to do this in the demo site and print a label like this

![image](https://github.com/inventree/InvenTree/assets/26359077/25dedee2-25ba-44bf-b7f6-8a6df24fb5e6)

I did take a look here https://docs.inventree.org/en/latest/report/labels/#extend-base-template but not sure how to add these templates to my installation?

thanks

**Comment by yodaphone** [comment]
NM, figured it out. I assumed that the sample dataset will also include the label templates, which was not the case. I created one in the admin interface and I am good to go.

I have 1 issue though. 
The part has location information but when i created a label to include it in the label, it doesnt print the location

```
{% extends "label/label_base.html" %}

{% load l10n i18n barcode %}

{% block style %}

.qr {
    position: absolute;
    left: 0mm;
    top: 0mm;
    {% localize off %}
    height: {{ height }}mm;
    width: {{ height }}mm;
    {% endlocalize %}
}

.part {
    font-family: Arial, Helvetica, sans-serif;
    display: flex;
    position: absolute;
    {% localize off %}
    left: {{ height }}mm;
    {% endlocalize %}
    top: 2mm;
}

{% endblock style %}

{% block content %}

<img class='qr' alt='{% trans "QR code" %}' src='{% qrcode qr_data %}'>

<div class='part'>
    {{ part.full_name }}
    {{location.name}}
</div>

{% endblock content %}
```

I also tried the sample provide here https://docs.inventree.org/en/latest/report/labels/#label-templates

But the label prints only the QRCode & not the location info
![image](https://github.com/inventree/InvenTree/assets/26359077/aa79627a-0315-4885-a104-6f02a6bec67b)

![image](https://github.com/inventree/InvenTree/assets/26359077/31e54a1e-fd9b-4bf4-af4c-7c7146587ea5)

**Reply by SchrodingersGat** [reply]
@yodaphone by default, the "sample" reports should all be available on a new installation. This sounds like a bug. 

Can you please provide information on the version of InvenTree you are running?

**Comment by yodaphone** [comment]
Sure, here you go, please let me know if you need 

![image](https://github.com/inventree/InvenTree/assets/26359077/1619b34b-bef5-400f-bc7b-3a7d149a7178)

**Comment by SchrodingersGat** [comment]
Ok, that version definitely should have labels auto-created on server start. This is all that happens on the demo instance in any case.

Can you try restarting your server, and see if they appear?

**Reply by yodaphone** [reply]
I upgraded it to 0.15.2 & they showed up . thanks
